src/vse_sync_pp/analyzers/analyzer.py

```diff
@@ -265,3 +270,193 @@ def explain(self, data):
+                                     100, 200, 300, 400, 500, 600, 700, 800, 900, 1000, 2000, 3000,
+                                     4000, 5000, 6000, 7000, 8000, 10000])
+        # observation window upper bound limited to 100k samples in 5k increments
+        taus_above_tenkey = np.arange(10001, 100000, 5000)
+        # `taus_list` contains range limit of obervation window intervals for which to compute the statistic
+        self._taus_list = np.concatenate((taus_below_tenkey, taus_above_tenkey))
+        self._rate = None
+        self._lpf_signal = None
+        # atributtes to be initialized in derived class
+        self._accuracy = None
+
+    def calculate_limit(self, tau):
+        # return limit based on `tau`
+        if self._accuracy is None:
+            raise AttributeError('atributte not initialized')
+        for (low, high), f in self._accuracy.items():
+            if ((low is None or tau > low) and (tau <= high)):
+                return f(tau) * (self._limit / 100)
+
+    def out_of_range(self, taus, samples):
+        # return true if any value in `tdevs` is out of range
+        for tau, sample in zip(taus, samples):
+            mask = self.calculate_limit(tau)
+            if mask <= sample:
+                return True
+        return False
+
+    def prepare(self, rows):
+        idx = 0
+        try:
+            tstart = rows[0].timestamp + self._transient
+        except IndexError:
+            pass
+        else:
+            while idx < len(rows):
+                if tstart <= rows[idx].timestamp:
+                    break
+                idx += 1
+        return super().prepare(rows[idx:])
+
+    def calculate_filter(self, input_signal):
+        """Calculate digital low-pass filter from `input_signal`
+
+        scipy_signal.butter input arguments:
+          order of the filter: 1
+          critical Frequency:
+            cutoff Frequency: 0.1Hz
+            for digital filters is normalized from 0 to 1 where 1 is Nyquist freqency.
+          btype is band type or type of filter
+          analog=False since it is always a digital filter
+        scipy_signal.butter return arguments:
+          `numerator` coefficient vector and `denominator coefficient vector of the butterworth digital filter
+        """
+        numerator, denominator = scipy_signal.butter(1, 0.1 / (1 / 2), btype="low", analog=False, output="ba")
+        lpf_signal = scipy_signal.filtfilt(numerator, denominator, input_signal.terror)
+        lpf_signal = lpf_signal[self._transient:len(lpf_signal)]
+        return lpf_signal
+
+    def calculate_rate(self, data):
+        # calculate sample rate using 100 samples
+        cumdelta = 0
+        for i in range(1, 100):
+            cumdelta = cumdelta + data.iloc[i].timestamp - data.iloc[i - 1].timestamp
+        return round((1 / (cumdelta / 100)))
+
+    def test_common(self, data):
```

Comment:
```suggestion
    def _test_common(self, data):
```
Prefix with `_` to indicate it's intended for internal use.